### DISCUSSION TITLE
printing body when unexpected status code

### DISCUSSION THREAD

**Discussion by fpeterschmitt** [gavv/httpexpect#137] [Ideas]
Hi 👋 

I would like to know if you would be interested in having a new `Expect().Status(x)` behavior that would print out the response body if the status is not as expected.

My motivation is that i want to write scenarios instead of unitary endpoint tests. And for that, i need to use the "Require" reporter that will stop immediately after the first error encountered, otherwise the results are completely broken after the first failure :)

I have an idea for implementation: through configuration, being able to provide a custom status formatter/checker instead of the default, current implementation. This formatter/checker would be called after the chain check and receive the response as parameter.

What do you think?

**Comment by gavv** [comment]
Hi, I think this makes sense and actually we already have an issue for a very similar feature: #69. Will it solve your problem? The issue is still in progress though.

**Comment by fpeterschmitt** [comment]
yep that looks absolutely great! :)

**Comment by gavv** [comment]
This feature is not implemented yet, however #69 is finished and now httpexpect allows to implement it in user code. See https://github.com/gavv/httpexpect#customize-failure-formatting and https://github.com/gavv/httpexpect#customize-assertion-handling

Also I will create an issue for implementing this functionality inside httpexpect.

**Comment by gavv** [comment]
I've created issue: https://github.com/gavv/httpexpect/issues/159
